Update styles for the reply-to-review form

kumar303 opened the pull request:
Fixes https://github.com/mozilla/addons-frontend/issues/3370

- I left the text-shadow in the buttons even though the mock doesn't have it
- I didn't implement the preview-reply part because that's a lot of effort--it can come later
- The mock uses `$grey-50` (I think?) for the Cancel button which is why I didn't go with the `$neutral-base-color`
- I didn't add the placeholder guidelines (tips for writing a review) since that needs some product guidance. I'm not sure what tips we should be giving.

<img width="960" alt="screenshot 2017-10-09 09 50 19" src="https://user-images.githubusercontent.com/55398/31344417-41ef0812-acd8-11e7-9fa5-e8fbdace75b6.png">
<img width="959" alt="screenshot 2017-10-09 09 50 01" src="https://user-images.githubusercontent.com/55398/31344421-45587f24-acd8-11e7-91e4-48c24782d447.png">

<img width="320" alt="screenshot 2017-10-06 18 54 58" src="https://user-images.githubusercontent.com/55398/31302600-89d86d2e-aac8-11e7-8832-ff03154113bb.png">



This is what the centered buttons looked like (this change got reverted):


<img width="960" alt="screenshot 2017-10-06 18 54 06" src="https://user-images.githubusercontent.com/55398/31302597-83b23ff6-aac8-11e7-8f84-e8f2ebd48465.png">

<img width="959" alt="screenshot 2017-10-06 18 54 37" src="https://user-images.githubusercontent.com/55398/31302599-8781e550-aac8-11e7-86bc-9b61493d250b.png">

tofumatt commented:
The text shadow in the buttons is site-wide and from our use of pre-photon colours; without the shadow the buttons had accessibility problems because of their text size.

We should be able to fix that in #3398, which I've filed as a sort of tracker issue post-MVP.

tofumatt commented on src/ui/components/Button/Button.scss:
```diff
@@ -30,6 +30,15 @@
 }
 
+// This is intended for counter-actions like cancelling or undoing, etc.
+.Button--counter-action {
```
I get the notion that this is "counter" to the original action but I think that's a bit of unclear phrasing, especially as people might read this as a number "Counter" (which could also have a button).

"cancel" or "undo" would be a better name.

kumar303 replied:
I think counter-action is more versatile since it's generic enough to convey the concept of cover cancel, dismiss or undo but I can change it, no problem.

tofumatt commented on src/ui/components/DismissibleTextForm/styles.scss:
```diff
@@ -1,30 +1,64 @@
+    margin-top: 24px;
+
+    .Button {
+      margin-left: 48px;
```
This should use `@include margin-start(48px);` so it works in RTL as well.

tofumatt commented on src/ui/components/DismissibleTextForm/styles.scss:
```diff
@@ -1,30 +1,64 @@
+
+    .Button {
+      margin-left: 48px;
+      padding: 0 64px;
```
Is this supposed to centre the buttons? In the mocks it looks like they should be centered... though I'd argue that's worse for usability. Hacing them be farther away from each other prevents accidental taps (mainly on touch devices) and having them be on the edges of the screen makes it easier to tap when on a phone/tablet.

I see in the mocks it's:
<img width="376" alt="screenshot 2017-10-08 13 58 34" src="https://user-images.githubusercontent.com/90871/31317006-f0af1b56-ac30-11e7-98b8-7432b40beb52.png">

I also just think it looks a bit off. @pwalm: Could we align theses on the left/right instead of the centre?

kumar303 replied:
This adds left and right padding to make the buttons wide like the mocks have them.

tofumatt commented on src/ui/components/DismissibleTextForm/styles.scss:
```diff
@@ -1,30 +1,64 @@
+      padding: 0 64px;
+
+      &:first-child {
+        margin-left: 0;
```
Should be `margin-start` as well.

kumar303 commented:
@tofumatt I think the edge buttons look weird on a wide screen but I'm not picky so I changed them for you and updated the screenshots. It is ready for another review.

tofumatt commented on src/ui/components/DismissibleTextForm/styles.scss:
```diff
@@ -1,30 +1,58 @@
+    margin-top: 24px;
+
+    .Button {
+      padding: 0 64px;
```
I see now this is for making the button bigger.

I think this should be a button option (something like `Button--large` similar to our existing `Button--small`). I know it's a pain but I have seen a lot of code bases with tiny variances on buttons because they are modified in a non-central place.

It's fine to do it as a follow-up bug if it's a pain here, but we should at least have it filed (and maybe linked to #3398) to have a large button style any component can use rather than individual components changing button styles. It's fine to align/position buttons here, but I think anything we do *inside* the button should be in the button style as an option.

kumar303 replied:
Good call, I addressed it. I was hesitating before only because I didn't know if we wanted to apply the style globally or not. Using a new button style is a safe approach because it won't be applied to any existing buttons.
